### PR TITLE
Replace full-index-replace with merge strategy

### PR DESCRIPTION
**Pull request by julienvincent**
Currently when performing a web scrape the strategy is to pipe records into a `tmp` index and then replace the target index with the `tmp` one. This is a problem for us as we want to perform scrapes on subsets of a website and then merge the results into the target index.

The reason we want to do this as opposed to full website scrapes is to keep costs to a minimum. A full website scrape consumes around `9k` index operations while the entry tier pricing is capped at `100k` - meaning we would only be able to do 10 full website scrapes per month. Not great.

By doing scrapes of a subset of the site, we can drop that to an average of 20 - 200 `iops` per scrape. Much cheaper.

[ch6678](https://app.clubhouse.io/journeyapps/story/6678/dock-v1)

## Todo

- [ ] Replace CircleCI environment variables with generic platform ones (instead of personal `julien@journeyapps.com` creds)

### CODE REVIEW

**Comment by julienvincent** [comment]
[ch6678](https://app.clubhouse.io/journeyapps/story/6678/dock-v1)

File: scraper/src/algolia_helper.py [hunk 1]
```diff
@@ -39,15 +45,16 @@ def add_synonyms(self, synonyms):
...
         print(
             "\033[94m> DocSearch: \033[0m" + "Synonyms" + " (\033[93m" + str(
                 len(synonyms_list)) + " synonyms\033[0m)")
 
     def commit_tmp_index(self):
         """Overwrite the real index with the temporary one"""
         # print("Update settings")
-        self.algolia_client.move_index(self.index_name_tmp, self.index_name)
+        
+        # self.algolia_client.move_index(self.index_name_tmp, self.index_name)
```
Comment (dylanvorster) [inline scraper/src/algolia_helper.py]: still needed?

Comment (julienvincent) [reply scraper/src/algolia_helper.py]: no

File: server/server.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      });
+    } catch (err) {
+      res.writeHead(500);
+      res.write('error parsing body');
+      res.end();
+    }
+  });
+});
+
+server.listen(9989, () => {
```
Comment (dylanvorster) [inline server/server.js]: can we make this process.ENV.PORT || 9989

File: server/server.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  console.log(`server listening on port 9989`);
```
Comment (dylanvorster) [inline server/server.js]: and then obv update this

File: server/server.js (hunk 2 above)
Comment (julienvincent) [reply server/server.js]: we could - but I mean this is only ever used within a docker-container, and so you can control this with `docker run -p 1234:9989`

**Pull request merged**